### PR TITLE
add dist transpiler unit test

### LINKED ISSUE #10375: Add unit test for the distributed transpiler. Need to verify the transpiled program has desired output.

### PR DESCRIPTION
**Pull request by Yancey1989**
Fixed #10375

### CODE REVIEW

File: python/paddle/fluid/transpiler/common.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
+#   Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
```
Comment (typhoonzero) [inline python/paddle/fluid/transpiler/common.py]: The name is `common.py` while it's not quite "common", see https://github.com/PaddlePaddle/Paddle/issues/9770#issuecomment-379840606 for details

Comment (Yancey1989) [reply python/paddle/fluid/transpiler/common.py]: Thanks! `common.py` is only used by `distribute_transpiler`, so I delete it and reuse it for future.

File: python/paddle/fluid/tests/unittests/test_dist_transpiler.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+                        [p[0] for p in params_grads])
+
+        delete_ops(trainer.global_block(), optimize_ops)
+        return trainer
+
+    def get_trainer(self):
+        return self._transpiler_instance().get_trainer_program()
+
+    def get_pserver(self, ep):
+        return self._transpiler_instance().get_pserver_program(ep)
```
Comment (typhoonzero) [inline python/paddle/fluid/tests/unittests/test_dist_transpiler.py]: transpile will be called twice, we need to let `_transpiler_instance` call `t.transpile` only once.

File: python/paddle/fluid/tests/unittests/test_dist_transpiler.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        self.assertEqual([op.type for op in trainer.global_block().ops],
+                         [op.type for op in expect_trainer.global_block().ops])
+        self.assertEqual(len(pserver.blocks), 3)
+        # block0: listen_and_serv
+        self.assertEqual([op.type for op in pserver.blocks[0].ops],
+                         ["listen_and_serv"])
+        # block2: optimize pass
+        self.assertEqual([op.type for op in pserver.blocks[2].ops],
+                         ["sum", "scale", "sgd"])
+
```
Comment (typhoonzero) [inline python/paddle/fluid/tests/unittests/test_dist_transpiler.py]: startup also need to be checked.

File: python/paddle/fluid/tests/unittests/test_dist_transpiler.py (hunk 2 above)
Comment (Yancey1989) [reply python/paddle/fluid/tests/unittests/test_dist_transpiler.py]: May not, after call `get_trainer_program()`, the program which inside transpiler would be changed.

File: python/paddle/fluid/tests/unittests/test_dist_transpiler.py (hunk 3 above)
Comment (Yancey1989) [reply python/paddle/fluid/tests/unittests/test_dist_transpiler.py]: done.

Comment (typhoonzero) [reply python/paddle/fluid/tests/unittests/test_dist_transpiler.py]: These asserts only test that the op type is equal, I think that you man assume that the op's inputs and outputs are verified when creating the program which will run `InferShape`

**Pull request merged**